timelock-server/src/main/java/com/palantir/atlasdb/timelock/lock/LeaseExpirationTimer.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.lock;
+
+import com.palantir.common.time.Clock;
+
+public class LeaseExpirationTimer {
+
+    public static final long LEASE_TIMEOUT_MILLIS = 20_000;
+
+    private volatile long lastRefreshTimeMillis;
```

Comment:
Yeah this case is definitely possible. This doesn't affect correctness (since clients have to deal with locks becoming invalid at any time), but could result in a request successfully refreshing a lock token, and then us immediately expiring it. My initial thought was that in practice this would be so rare that it's not worth addressing this case - but open to other opinions here.